src/main.rs

```diff
@@ -10,6 +10,14 @@ fn main() {
     let mut file = File::open("./src/example.gpl").unwrap();
     let mut contents = vec![];
     file.read_to_end(&mut contents).unwrap();
+    // this removes \r as it seems to cause problems during parsing
+    for i in 0..contents.len() {
+        let c = contents.get(i);
+        match c {
+            Some(&13) => {contents.remove(i);},
+            _ => ()
+        }
```

Comment:
I think this code can be a bit cleaner:
```suggestion
        if let Some(&13) = contents.get(i) {
            contents.remove(i);
        }
```